src/Library/KubernetesCorsPolicyService.cs

```diff
@@ -10,7 +12,21 @@ namespace Contrib.IdentityServer4.KubernetesStore
     public class KubernetesCorsPolicyService : InMemoryCorsPolicyService
     {
         public KubernetesCorsPolicyService(ILogger<KubernetesCorsPolicyService> logger, ICustomResourceWatcher<ClientResource> clientWatcher)
-            : base(logger, clientWatcher.RawResources.Select(resource => resource.Spec))
-        {}
+            : base(logger, clientWatcher.RawResources.Select(resource => resource.Spec).Where(client => HasWellFormedAllowedCorsOrigins(client, logger)))
+        { }
+
+        private static bool HasWellFormedAllowedCorsOrigins(Client client, ILogger logger)
+        {
+            var hasWellFormedAllowedCorsOrigins = client.AllowedCorsOrigins.All(origin => Uri.IsWellFormedUriString(origin, UriKind.RelativeOrAbsolute));
+            if (hasWellFormedAllowedCorsOrigins)
+            {
+                return true;
+            }
+            else
+            {
+                logger.LogCritical($"Identity Client with name {client.ClientName} has invalid AllowedCorsOrigins");

```

Comment:
Critical means that the service is forced to shut down immediately to prevent data loss or corruption. Besides that, it's not an error of the service itself. Instead, `LogWarning` should be used.
```suggestion
                logger.LogWarning($"Identity Client with name {client.ClientName} has invalid AllowedCorsOrigins");
```